### PR TITLE
KOGITO-3563 Force Process definition beans to created once Quarkus ap…

### PR DESCRIPTION
**Pull request by cristianonicolai**
…p starts

### CODE REVIEW

**Review by evacchi — APPROVED** [review]

**Review by Sgitario — COMMENTED** [review]
I'm ok with the changes, but is there any test we could add to ensure the issue https://issues.redhat.com/browse/KOGITO-3563 is fixed?

**Review by fjtirado — APPROVED** [review]
LGTM

**Comment by cristianonicolai** [comment]
@Sgitario I'm not sure really how I could reliably test that ( app does not start without Infinispan connection ), open to suggestions :)

**Review by Sgitario — APPROVED** [review]
lgtm

**Review by ricardozanini — APPROVED** [review]
I liked it.

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/di/SpringDependencyInjectionAnnotator.java [hunk 1]
```diff
@@ -162,4 +162,9 @@ public String objectMapperInjectorSource(String packageName) {
         clazz.setPackageDeclaration( packageName );
         return clazz.toString();
     }
+
+    @Override
+    public <T extends NodeWithAnnotations<?>> T withEagerStartup(T node) {
+        return node;
```
Comment (MarianMacik) [inline kogito-codegen/src/main/java/org/kie/kogito/codegen/di/SpringDependencyInjectionAnnotator.java:168]: Maybe a short JavaDoc that it is no-op for Spring Boot would be welcome here.

**Review by MarianMacik — APPROVED** [review]

File: kogito-codegen/src/main/java/org/kie/kogito/codegen/di/SpringDependencyInjectionAnnotator.java (hunk 1 above)
Comment (cristianonicolai) [reply kogito-codegen/src/main/java/org/kie/kogito/codegen/di/SpringDependencyInjectionAnnotator.java:173]: added Javadoc

**Pull request merged**